errdef/errors.go

```diff
@@ -19,12 +19,13 @@ import "errors"
 
 // Common errors used in ORAS
 var (
-	ErrAlreadyExists      = errors.New("already exists")
-	ErrUnsupported        = errors.New("unsupported")
-	ErrInvalidReference   = errors.New("invalid reference")
-	ErrInvalidDigest      = errors.New("invalid digest")
-	ErrNotFound           = errors.New("not found")
-	ErrUnsupportedVersion = errors.New("unsupported version")
-	ErrMissingReference   = errors.New("missing reference")
-	ErrSizeExceedsLimit   = errors.New("size exceeds limit")
+	ErrAlreadyExists            = errors.New("already exists")
+	ErrUnsupported              = errors.New("unsupported")
+	ErrInvalidReference         = errors.New("invalid reference")
+	ErrInvalidReferenceTerminal = errors.New("terminally invalid reference")
```

Comment:
In `Repository.ParseReference`, the error `InvalidReference` signals an error, but one where a recovery is attempted (`if err != nil { // recovery }`.  This new error type lets the caller know that a recovery should not be attempted, and this is a terminal state.

Reply:
How do you define `ErrInvalidReferenceTerminal`? Users will be confused with `ErrInvalidReference` and `ErrInvalidReferenceTerminal`.

We should only have one error. That is, `ErrInvalidReference`.

Reply:
`ErrInvalidReferenceTerminal` has been removed.